InvenTree/part/api.py

```diff
@@ -499,6 +503,9 @@ def filter_queryset(self, queryset):
         # Filter by "parts which need stock to complete build"
         stock_to_build = params.get('stock_to_build', None)
 
+        # TODO: This is super expensive, database query wise...
+        # TODO: Need to figure out a cheaper way of making this filter query
+
         if stock_to_build is not None:
```

Comment:
@eeintech this approach is pretty "expensive" in terms of database hits. It runs into the [N+1 query problem](https://scoutapm.com/blog/django-and-the-n1-queries-problem) as the query cannot be performed in a single run.

This will need some further investigation to determine if it can be sped up

Reply:
@SchrodingersGat Right, I wasn't in love with this implementation, I'll have to think about this more...

Reply:
I just realized `django-debug-toolbar` can be enabled in the config, I'll check it out!

Reply:
Yeah it is a good way to see how many SQL queries are actually being executed (among other things too)